x/blob/types/params.go

```diff
@@ -0,0 +1,99 @@
+package types
+
+import (
+	"fmt"
+
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"gopkg.in/yaml.v2"
+)
+
+var _ paramtypes.ParamSet = (*Params)(nil)
+
+var (
+	KeyMinSquareSize            = []byte("MinSquareSize")
+	DefaultMinSquareSize uint32 = 1
+)
+
+var (
+	KeyMaxSquareSize            = []byte("MaxSquareSize")
+	DefaultMaxSquareSize uint32 = 128
+)
+
+// ParamKeyTable returns the param key table for the blob module
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+// NewParams creates a new Params instance
+func NewParams(
+	minSquareSize uint32,
+	maxSquareSize uint32,
+) Params {
+	return Params{
+		MinSquareSize: minSquareSize,
+		MaxSquareSize: maxSquareSize,
+	}
+}
+
+// DefaultParams returns a default set of parameters
+func DefaultParams() Params {
+	return NewParams(
+		DefaultMinSquareSize,
+		DefaultMaxSquareSize,
+	)
+}
+
+// ParamSetPairs gets the list of param key-value pairs
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyMinSquareSize, &p.MinSquareSize, validateMinSquareSize),
+		paramtypes.NewParamSetPair(KeyMaxSquareSize, &p.MaxSquareSize, validateMaxSquareSize),
+	}
+}
+
+// Validate validates the set of params
+func (p Params) Validate() error {
+	if err := validateMinSquareSize(p.MinSquareSize); err != nil {
+		return err
+	}
+
+	if err := validateMaxSquareSize(p.MaxSquareSize); err != nil {
+		return err
+	}
+
```

Comment:
Probably also add a `minSquareSize > maxSquareSize` check and error in that case. Not sure if this is covered somewhere else tho.